carmm/run/aims_calculator.py

```diff
@@ -21,29 +21,44 @@ def get_aims_calculator(dimensions, k_grid=None, xc="pbe", compute_forces="true"
             FHI_calc: FHI-aims ASE calculator
        
     '''
-
-    from ase.calculators.aims import Aims
+    # Changing to check ASE version, as this determines behaviour of calculator
+    from carmm.utils.python_env_check import ase_env_check
+    if ase_env_check('3.23.0'):
+        from ase.calculators.aims import Aims, AimsProfile
+    else:
+        from ase.calculators.aims import Aims, Aims as AimsProfile
 
     # Default is suitable for molecular calculations
-    fhi_calc = Aims(
-        spin='none',
-        relativistic=('atomic_zora', 'scalar'),
-        compute_forces=compute_forces,
-        **kwargs
-    )
 
     # Set the XC for the calculation. For LibXC, override_warning_libxc *needs*
     # to be set first, otherwise we get a termination.
+
+    # Created dictionary to store arguments
+    parameter_dict = {}
     if "libxc" in xc:
-        fhi_calc.set(override_warning_libxc="true")
-    fhi_calc.set(xc=xc)
+        parameter_dict['override_warning_libxc'] = 'true'
+    parameter_dict['xc'] = xc
 
     if dimensions == 2:
-        fhi_calc.set(use_dipole_correction='true')
+        parameter_dict['use_dipole_correction'] = 'true'
 
     if dimensions >= 2:
-        fhi_calc.set(k_grid=k_grid)
+        parameter_dict['k_grid'] = k_grid
 
+    if spin is not None:
+        parameter_dict['spin'] = 'none'
+
+    if relativistic is None:
+        parameter_dict['relativistic'] = ('atomic_zora', 'scalar')
+
+    fhi_calc = Aims(
+        profile=AimsProfile(command='xc'),
```

Comment:
Hi Andy. I am not sure about this line. My understanding is this.
If ase version is 3.22, `AimsProfile(command='xc')` is an Aims calculator. Will this pollute the `control.in` file `fhi_calc` generates? I didn't find where they remove the non-existing parameters in the old write_control function.
If ase version is 3.23, `AimsProfile(command='xc')` provides a profile for `fhi_calc`. This seems to let dry run work. But does it prevent the `fhi_calc` to load commands and paths when doing real calculations?

Reply:
Fair questions, but hard to answer as you are basically asking for situations outside the boundaries of our CI tests. The only way we can find out is by testing, and then improving the CI tests if things don't work.

Reply:
For actual use with ASE v3.23, all we have currently is the CI tests - we'd have to deploy and use to see if it works, or doesn't, and then resolve as necessry.

Reply:
(We can't fix bugs we don't know about!)

Reply:
@GaryLZW is this at least clearer now in implementation? And we can work to see how things work when actually deployed?

Reply:
Hi Andy. Yes. The code looks clearer to me now. Happy to resolve further issues after deploying.

Reply:
I think a more permanent solution should be feasible when we get around to re-working ReactAims and related calculator handling routines for ASE 3.23 - I can imagine we can create a dummy profile that is loaded for CI testing, and loads the user-generated profile otherwise.

Reply:
Agreed - there is a better solution out there. I think we also need to make the executable commands internal, rather than system environment settings.